agent/api/task/task.go

```diff
@@ -302,7 +302,13 @@ func TaskFromACS(acsTask *ecsacs.Task, envelope *ecsacs.PayloadMessage) (*Task,
 		task.StopSequenceNumber = *envelope.SeqNum
 	}
 
-	// Overrides the container command if it's set
+	// get service connect config value from ACS attachment
+	scConfig, err := ExtractSCConfigFromACSAttachment(acsTask)
+	if err != nil {
+		return nil, err
+	}
+	task.ServiceConnectConfig = scConfig
```

Comment:
This is outside the scope of this review. But why do we want Task to hold the SC Config. Could we move the initialization of ephemeral ports,etc into `sevice_connect.go`?

Reply:
> But why do we want Task to hold the SC Config

SC Config is an attribute of the task, and as such it will be persisted/loaded from to the agent database in case of agent restarts.

> Could we move the initialization of ephemeral ports, etc into sevice_connect.go?'

Yeah, I think we could move all that to service connect in  a new PR.

Reply:
Yes. Lets take it up in a new PR.